File: files/CH-189.R
library(tidyverse)
library(readxl)

path = "CH-189 Combining the columns.xlsx"
input = read_excel(path, range = "B2:E7")
test  = read_excel(path, range = "H2:H7")

result = input %>%
  mutate(Pattern = trimws(Pattern)) %>%
  separate_rows(Pattern, sep = "") %>%
  mutate(repl = case_when(
    Pattern == "F" ~ `First Name`,
    Pattern == "L" ~ `Last Name`,
    Pattern == "M" ~ `Middle Name`,
    TRUE ~ Pattern
  )) %>%
  summarise(`Custom Format` = paste(repl, collapse = ""), .by = `First Name`)

File: files/CH-189.py
import pandas as pd

path = "CH-189 Combining the columns.xlsx"
input = pd.read_excel(path, usecols="B:E", skiprows=1, nrows=6)

def replace_pattern(row):
    return ''.join(row['First Name'] if c == 'F' else row['Last Name'] if c == 'L' else row['Middle Name'] if c == 'M' else c for c in row['Pattern'].strip())

input['Custom Format'] = input.apply(replace_pattern, axis=1)
result = input.groupby('First Name')['Custom Format'].apply(''.join).reset_index()

print(result)
